proof/invariant-abstract/AARCH64/ArchDetSchedAux_AI.thy

```diff
@@ -15,67 +15,27 @@ named_theorems DetSchedAux_AI_assms
 
 crunch init_arch_objects
   for exst[wp]: "\<lambda>s. P (exst s)"
-  and valid_etcbs[wp, DetSchedAux_AI_assms]: valid_etcbs
   and valid_queues[wp]: valid_queues
   and valid_sched_action[wp]: valid_sched_action
   and valid_sched[wp]: valid_sched
-  (wp: mapM_x_wp')
-
-(* already proved earlier *)
-declare invoke_untyped_cur_thread[DetSchedAux_AI_assms]
-
-crunch invoke_untyped
-  for ready_queues[wp, DetSchedAux_AI_assms]: "\<lambda>s. P (ready_queues s)"
-  and scheduler_action[wp, DetSchedAux_AI_assms]: "\<lambda>s. P (scheduler_action s)"
+  and etcbs_of[wp, DetSchedAux_AI_assms]: "\<lambda>s. P (etcbs_of s)"
```

Comment:
Does `etcbs_of` still exist when we removed the `ekheap`?

Reply:
Yes, to minimise the changes required we still define it in `DetSchedInvs_AI`, as a projection of `kheap`. This is the same as what was done in `rt`.